cli/deal/showdeals.go

```diff
@@ -0,0 +1,86 @@
+package deal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	lapi "github.com/filecoin-project/lotus/api"
+)
+
+var mockDealInfos []lapi.DealInfo
+
+func InspectDealCmd(ctx context.Context, api lapi.FullNode, proposalCid string, dealId int) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	deals, err := api.ClientListDeals(ctx)
+	if err != nil {
+		return err
+	}
+
+	var di lapi.DealInfo
+	found := false
+	for _, cdi := range deals {
+		if proposalCid != "" && cdi.ProposalCid.String() == proposalCid {
+			di = cdi
+			found = true
+			break
+		}
+
+		if dealId != 0 && int(cdi.DealID) == dealId {
+			di = cdi
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		if proposalCid != "" {
+			return fmt.Errorf("cannot find deal with proposal cid: %s", proposalCid)
+		}
+		if dealId != 0 {
+			return fmt.Errorf("cannot find deal with deal id: %v", dealId)
+		}
+		return errors.New("you must specify proposal cid or deal id in order to inspect a deal")
+	}
+
+	renderDeal(di)
+
+	return nil
+}
+
+func renderDeal(di lapi.DealInfo) {
+	color.Blue("Deal ID:      %d\n", int(di.DealID))
+	color.Blue("Proposal CID: %s\n\n", di.ProposalCid.String())
+
+	for _, stg := range di.DealStages.Stages {
+		msg := fmt.Sprintf("%s %s: %s (%s)", color.BlueString("Stage:"), color.BlueString(strings.TrimPrefix(stg.Name, "StorageDeal")), stg.Description, color.GreenString(stg.ExpectedDuration))
```

Comment:
Ideally we'd do a `fmt.Fprintf` in the `cli.Context.App.Writer`, but if all other commands are writing outputs directly to stdout, then we can keep it like it is.